src/ObjectGraph.es6.js

```diff
@@ -224,7 +250,9 @@ ObjectGraph.prototype.visitPropertyDescriptors = function(o, metadataMap) {
 // Visit an object, o. Return an id for the object, which may contain type
 // information (e.g., number, boolean, null), or indicate the unique identity
 // of the object itself.
-ObjectGraph.prototype.visitObject = function(o) {
+ObjectGraph.prototype.visitObject = function(o, opt) {
```

Comment:
Comment added. Opt stores some additional information of this object.
opt.proto contains if this object is visited through another object's __proto__.